src/main/java/com/datastax/cdm/job/SplitPartitions.java

```diff
@@ -168,13 +168,32 @@ private static BufferedReader getfileReader(String fileName) {
         }
     }
 
-    public static String getPartitionFile(PropertyHelper propertyHelper) {
-        String filePath = propertyHelper.getString(KnownProperties.TOKEN_RANGE_PARTITION_FILE);
-        if (StringUtils.isAllBlank(filePath)) {
-            filePath = "./" + propertyHelper.getString(KnownProperties.ORIGIN_KEYSPACE_TABLE) + "_partitions.csv";
+    public static boolean appendPartitionOnDiff(PropertyHelper propertyHelper) {
+        return Boolean.TRUE.equals(propertyHelper.getBoolean(KnownProperties.TOKEN_RANGE_PARTITION_FILE_APPEND_ON_DIFF));
+    }
+
+    public static String getPartitionFileInput(PropertyHelper propertyHelper) {
+        if (!StringUtils.isAllBlank(propertyHelper.getString(KnownProperties.TOKEN_RANGE_PARTITION_FILE_INPUT))) {
+            return propertyHelper.getString(KnownProperties.TOKEN_RANGE_PARTITION_FILE_INPUT);
+        }
+
+        if (!StringUtils.isAllBlank(propertyHelper.getString(KnownProperties.TOKEN_RANGE_PARTITION_FILE))) {
+            return propertyHelper.getString(KnownProperties.TOKEN_RANGE_PARTITION_FILE);
```

Comment:
I would recommend we remove this property & completely replace with the `input` & `output` properties you have. I understand this breaks backward compatibility, but it will be more clear. In most scenarios when this feature is used, users just go with the default option, which will still continue to work.